src/core/Global.ttslua

```diff
@@ -838,15 +848,33 @@ function onClick_toggleOption(_, id)
   applyOptionPanelChange(id, state)
 end
 
+-- called by the language selection dropdown
+function languageSelected(_, selectedIndex, id)
+  optionPanel[id] = LANGUAGES[tonumber(selectedIndex) + 1].name
+end
+
+function returnLanguageId(name)
+  for index, tbl in ipairs(LANGUAGES) do
+    for key, value in pairs(tbl) do
```

Comment:
You know what the elements in the table are, seems like you could just check name and return code rather than having the inner loop?  Or am I missing something (it's late)

Reply:
Good catch. Updated it